src/redux/slices/transactions/index.js

```diff
@@ -1,7 +1,103 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import set from 'lodash-es/set';
 import { createSelector } from 'reselect';
 
+import { getTransactions, transactionExtraInfo } from '../../../utils/explorer-api';
+
 const SLICE_NAME = 'transactions';
 
+const initialState = {
+    transactions: {
+        byAccountId: {}
+    },
+    status: {
+        
+    }
+};
+
+const fetchTransactions = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactions`,
+    async ({ accountId }, { dispatch, getState }) => {
+        const transactions = await getTransactions({ accountId });
+
+        const { actions: { addTransactions, updateTransactions } } = transactionsSlice;
+
+        !selectTransactionsByAccountId(getState(), { accountId }).length
+            ? dispatch(addTransactions({ transactions, accountId }))
+            : dispatch(updateTransactions({ transactions, accountId }));
+    }
+);
+
+const fetchTransactionStatus = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactionStatus`,
+    async ({ hash, signer_id, accountId }, { dispatch, getState }) => {
+        let status;
+        try {
+            const transactionDetails = await transactionExtraInfo({ hash, signer_id });
+            status = Object.keys(transactionDetails.status)[0];
+        } catch (error) {
+            status = 'notAvailable';
+        }
+        const checkStatus = ['SuccessValue', 'Failure'].includes(status);
+
+        const { actions: { updateTransactionStatus } } = transactionsSlice;
+        dispatch(updateTransactionStatus({ status, checkStatus, accountId, hash }));
+    }
+);
+
+const transactionsSlice = createSlice({
+    name: SLICE_NAME,
+    initialState,
+    reducers: {
+        addTransactions(state, { payload }) {
```

Comment:
"Add" usually means 'in addition to something that is already there', so this implied to me that it appended new transactions to the existing array.  What do you think about making this `setTransactions` -- `set`ing something generally means replacing whatever value is there, which is what this is doing

Reply:
Agree, done :)